components/apimgt/org.wso2.carbon.apimgt.rest.api.publisher.v1/src/main/java/org/wso2/carbon/apimgt/rest/api/publisher/v1/impl/ApisApiServiceImpl.java

```diff
@@ -695,6 +749,75 @@ public Response apisApiIdPut(String apiId, APIDTO body, String ifMatch, MessageC
         } catch (FaultGatewaysException e) {
             String errorMessage = "Error while updating API : " + apiId;
             RestApiUtil.handleInternalServerError(errorMessage, e, log);
+        } catch (CryptoException e) {
+            log.error("Error while encrypting the secret key", e);
```

Comment:
add some additional information to uniquely identify the error log. i:e user/tenantdomain/api